File: languages/c++/patterns/3 State/Context.cpp
#include "State.h"
#include "StateAppend1.h"
#include "Context.h"

Context::Context(){
	currentState = new StateAppend1;
}

void Context::setState(State* s){
	currentState = s;
}

void Context::write(string s){
	currentState->write(this, s);
}

File: languages/java/3 State/Main/src/main/Context.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package main;

/**
 *
 * @author lupin
 */
public class Context {
    private State currentState;
    public Context(){
        currentState = new StateAppend1();
    }
    public void setState(State s){
        currentState = s;
    }
    public void write(String s){
        currentState.write(this, s);
    }
}
